rules/prefer-key-over-key-code.js

```diff
@@ -0,0 +1,189 @@
+'use strict';
+const getDocsUrl = require('./utils/get-docs-url');
+
+const keys = ['keyCode', 'charCode', 'which'];
+// https://github.com/facebook/react/blob/b87aabd/packages/react-dom/src/events/getEventKey.js#L36
+// only meta characters which can't be deciphered from String.fromCharCode
+const translateToKey = {
+	8: 'Backspace',
+	9: 'Tab',
+	12: 'Clear',
+	13: 'Enter',
+	16: 'Shift',
+	17: 'Control',
+	18: 'Alt',
+	19: 'Pause',
+	20: 'CapsLock',
+	27: 'Escape',
+	32: ' ',
+	33: 'PageUp',
+	34: 'PageDown',
+	35: 'End',
+	36: 'Home',
+	37: 'ArrowLeft',
+	38: 'ArrowUp',
+	39: 'ArrowRight',
+	40: 'ArrowDown',
+	45: 'Insert',
+	46: 'Delete',
+	112: 'F1',
+	113: 'F2',
+	114: 'F3',
+	115: 'F4',
+	116: 'F5',
+	117: 'F6',
+	118: 'F7',
+	119: 'F8',
+	120: 'F9',
+	121: 'F10',
+	122: 'F11',
+	123: 'F12',
+	144: 'NumLock',
+	145: 'ScrollLock',
+	224: 'Meta'
+};
+
+const isPropertyNamedAddEventListener = node =>
+	node &&
+	node.callee &&
+	node.callee.property &&
+	node.callee.property.name === 'addEventListener' &&
+	node;
+
+const isInsideEventCallback = ancestors => {
+	return ancestors.reduce((isInside, current) => {
+		return isInside || isPropertyNamedAddEventListener(current);
+	}, null);
+};
+
+const isPropertyOf = (node, objectName) => {
+	return (
+		node &&
+		node.parent &&
+		node.parent.object &&
+		node.parent.object.name === objectName
+	);
+};
+
+const getEventNode = node => {
+	switch (node && node.type) {
+		case 'ArrowFunctionExpression':
+		case 'FunctionExpression': {
+			return node.params && node.params[0] && node.params[0].name;
+		}
+
+		default:
+			return null;
+	}
+};
+
+const extraCitation = 'See https://goo.gl/cRK532 for more info.';
+
+const util = context => {
+	const getNearestAncestorByType = nodes => type => {
+		for (let i = nodes.length - 1; i >= 0; i--) {
+			if (nodes[i].type === type) {
+				return nodes[i];
```

Comment:
I've totally removed this implementation. Now traversing the parent instead of relying on ancestors, which I thought was a little misleading in case I wanted to check the parents of another node in the current traversal.